HW02LAZY/LAZY/Lazy.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Runtime.InteropServices;
+
+namespace MyLazy;
+
+public interface ILazy<T> { T? Get(); }
+
+/// <summary>
+/// Thread non-safe realisation of built-in Lazy class
+/// </summary>
+/// <typeparam name="T">Type of supplier function returning value</typeparam>
+public class LazySingleThread<T> : ILazy<T>
+{
+    private bool IsCalculated = false;
+    private T? Result;
+    private Func<T> Supplier;
```

Comment:
по стайлгайду приватные свойства и методы нужно называть с нижним подчёркиванием в начале и с маленькой буквы:
```suggestion
    private bool _isCalculated = false;
    private T? _result;
    private Func<T> _supplier;
```